airflow/www/templates/airflow/dags.html

```diff
@@ -524,10 +524,18 @@ <h2>DAGs</h2>
 
       if (encoded_dag_ids.length > 0) {
         // dags on page fetch stats
-        d3.json("{{ url_for('Airflow.blocked') }}?dag_ids="  + (encoded_dag_ids.join(',')), blockedHandler);
-        d3.json("{{ url_for('Airflow.last_dagruns') }}?dag_ids=" + (encoded_dag_ids.join(',')), lastDagRunsHandler);
-        d3.json("{{ url_for('Airflow.dag_stats') }}?dag_ids=" + (encoded_dag_ids.join(',')), dagStatsHandler);
-        d3.json("{{ url_for('Airflow.task_stats') }}?dag_ids=" + (encoded_dag_ids.join(',')), taskStatsHandler);
+        d3.json("{{ url_for('Airflow.blocked') }}")
+          .header("X-CSRFToken", "{{ csrf_token() }}")
+          .post("dag_ids="  + (encoded_dag_ids.join(',')), blockedHandler);
+        d3.json("{{ url_for('Airflow.last_dagruns') }}")
+          .header("X-CSRFToken", "{{ csrf_token() }}")
+          .post("dag_ids="  + (encoded_dag_ids.join(',')), lastDagRunsHandler);
+        d3.json("{{ url_for('Airflow.dag_stats') }}")
+          .header("X-CSRFToken", "{{ csrf_token() }}")
+          .post("dag_ids="  + (encoded_dag_ids.join(',')), dagStatsHandler);
+        d3.json("{{ url_for('Airflow.task_stats') }}")
+          .header("X-CSRFToken", "{{ csrf_token() }}")
+          .post("dag_ids="  + (encoded_dag_ids.join(',')), taskStatsHandler);
```

Comment:
Is this right? I would expect that this would post the encoded as a single parameter..

Reply:
We probably don't have to handle the encoding and building of forms ourselves:

```js
var p = new URLSearchParams();
p.append("dag_id", "not+encoded.here");
p.append("dag_id", "second dag/id");
d3.json("/a").post(p)
```

And that sends this request:

```
POST /a HTTP/1.1
Host: 0.0.0.0:8080
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:72.0) Gecko/20100101 Firefox/72.0
Accept: application/json,*/*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Content-Type: application/x-www-form-urlencoded;charset=UTF-8
Content-Length: 48
Origin: http://0.0.0.0:8080
Connection: keep-alive
Referer: http://0.0.0.0:8080/home

dag_id=not%2Bencoded.here&dag_id=second+dag%2Fid
```

Could you make this change @madison-ookla ?

Reply:
Sure, can do

Reply:
Done! I've tested this locally and it works 👍

Reply:
Ready for you now @ashb @mik-laj 😁